src/mlpack/core/dists/gamma_distribution.hpp

```diff
@@ -52,176 +52,178 @@ namespace distribution {
+   *
+   * @param data Data to train the distribution on.
+   * @param tol Convergence tolerance. This is *not* an absolute measure:
+   *    It will stop the approximation once the *change* in the value is
+   *    smaller than tol.
+   */
+  GammaDistribution(const arma::mat& data, const double tol = 1e-8);
+
+  /**
+   * Construct the Gamma distribution given two vectors alpha and beta.
+   *
+   * @param alpha The vector of alphas, one per dimension.
+   * @param beta The vector of betas, one per dimension.
+   */
+  GammaDistribution(const arma::vec& alpha, const arma::vec& beta);
+
+  /**
+   * Destructor.
+   */
+  ~GammaDistribution() {}
+
+  /**
+   * This function trains (fits distribution parameters) to new data or the
+   * dataset the object owns.
+   *
+   * @param rdata Reference data to fit parameters to.
+   * @param tol Convergence tolerance. This is *not* an absolute measure:
+   *    It will stop the approximation once the *change* in the value is
+   *    smaller than tol.
+   */
+  void Train(const arma::mat& rdata, const double tol = 1e-8);
+
+  /**
+   * Fits an alpha and beta parameter according to observation probabilities.
+   * This method is not yet implemented.
+   *
+   * @param observations The reference data, one observation per column.
+   * @param probabilities The probability of each observation. One value per
+   *     column of the observations matrix.
+   * @param tol Convergence tolerance. This is *not* an absolute measure:
+   *    It will stop the approximation once the *change* in the value is
+   *    smaller than tol.
+   */
+  void Train(const arma::mat& observations,
+             const arma::vec& probabilities,
+             const double tol = 1e-8);
+
+  /**
+   * This function trains (fits distribution parameters) to a dataset with
+   * pre-computed statistics logMeanx, meanLogx, meanx for each dimension.
+   *
+   * @param logMeanxVec Is each dimension's logarithm of the mean
+   *     (log(mean(x))).
+   * @param meanLogxVec Is each dimension's mean of logarithms
+   *     (mean(log(x))).
+   * @param meanxVec Is each dimension's mean (mean(x)).
+   * @param tol Convergence tolerance. This is *not* an absolute measure:
+   *    It will stop the approximation once the *change* in the value is
+   *    smaller than tol.
+   */
+  void Train(const arma::vec& logMeanxVec,
+             const arma::vec& meanLogxVec,
+             const arma::vec& meanxVec,
+             const double tol = 1e-8);
+
+
+  /**
+   * This function returns the probability of a group of observations.
+   *
+   * The probability of the value x is
+   *
+   * \frac{x^(\alpha - 1)}{\Gamma(\alpha) * \beta^\alpha} * e ^
+   * {-\frac{x}{\beta}}
+   *
+   * for one dimension. This implementation assumes each dimension is
+   * independent, so the product rule is used.
+   *
+   * @param observations Matrix of observations, one per column.
+   * @param probabilities column vector of probabilities, one per
```

Comment:
Please make the sentence starting after parameter argument begin with capital letter word. Here and below.